packages/core-web/src/media/browser/webrtc/shared.ts

```diff
@@ -106,8 +90,13 @@ export async function negotiateConnectionWithClientOffer(
         new RTCSessionDescription({ type: 'answer', sdp: answerSDP }),
       );
       const sdpLinkHeader = response.headers.get('Link');
-
-      return parseIceServersFromLinkHeader(sdpLinkHeader);
+      const iceServers = parseIceServersFromLinkHeader(sdpLinkHeader);
+      peerConnection.setConfiguration({
+        iceServers: iceServers,
+      });
+      const offer = await peerConnection.createOffer();
+      await peerConnection.setLocalDescription(offer);
```

Comment:
I don't think this is correct - I think the server and client will be out of sync if we create an entirely new offer and set it on the local connection. IIUC you need the server to be aware of the offer - we're basically throwing away the original offer and creating a new one. cc @Thulinma for feedback on this

Reply:
From Discord - this is actually to spec. Resolving.